src/utils/daos_hdlr.c

```diff
@@ -1792,59 +1890,63 @@ fs_copy(struct cmd_args_s *ap,
 	char		*tmp_name = NULL;
 
 	/* Make sure the source exists. */
-	rc = file_lstat(ap, src_file_dfs, src_path, &src_stat);
+	rc = file_lstat(copy_args->ap, copy_args->src_file_dfs, copy_args->src_path, &src_stat);
 	if (rc != 0) {
 		rc = daos_errno2der(rc);
-		DH_PERROR_DER(ap, rc, "Failed to stat '%s'", src_path);
+		DH_PERROR_DER(copy_args->ap, rc, "Failed to stat '%s'", copy_args->src_path);
 		D_GOTO(out, rc);
 	}
 
 	/* If the destination exists and is a directory,
 	 * copy INTO the directory instead of TO it.
 	 */
-	rc = file_lstat(ap, dst_file_dfs, dst_path, &dst_stat);
+	rc = file_lstat(copy_args->ap, copy_args->dst_file_dfs, copy_args->dst_path, &dst_stat);
 	if (rc == 0) {
 		if (S_ISDIR(dst_stat.st_mode)) {
 			copy_into_dst = true;
 		} else if S_ISDIR(src_stat.st_mode) {
 			rc = -DER_INVAL;
-			DH_PERROR_DER(ap, rc, "Destination is not a directory");
+			DH_PERROR_DER(copy_args->ap, rc, "Destination is not a directory");
 			D_GOTO(out, rc);
 		}
 	}
 
 	if (copy_into_dst) {
 		/* Get the dirname and basename */
-		rc = parse_filename_dfs(src_path, &tmp_name, &tmp_dir);
+		rc = parse_filename_dfs(copy_args->src_path, &tmp_name, &tmp_dir);
 		if (rc != 0) {
-			DH_PERROR_DER(ap, rc, "Failed to parse path '%s'", src_path);
+			DH_PERROR_DER(copy_args->ap, rc, "Failed to parse path '%s'",
+				      copy_args->src_path);
 			D_GOTO(out, rc);
 		}
 
 		/* Build the destination path */
 		if (tmp_name != NULL) {
-			D_ASPRINTF(tmp_path, "%s/%s", dst_path, tmp_name);
+			D_ASPRINTF(tmp_path, "%s/%s", copy_args->dst_path, tmp_name);
 			if (tmp_path == NULL)
 				D_GOTO(out, rc = -DER_NOMEM);
-			dst_path = tmp_path;
+			copy_args->dst_path = tmp_path;
 		}
 	}
 
 	switch (src_stat.st_mode & S_IFMT) {
 	case S_IFREG:
-		rc = fs_copy_file(ap, src_file_dfs, dst_file_dfs, &src_stat, src_path, dst_path);
+		rc = fs_copy_file(copy_args->ap, copy_args->src_file_dfs, copy_args->dst_file_dfs,
+				  &src_stat, copy_args->src_path, copy_args->dst_path);
 		if (rc == 0)
-			(*num_files)++;
+			(copy_args->num_files)++;
```

Comment:
I don't think you need the brackets here do you?  Same comment below.
```suggestion
			copy_args->num_files++;
```

Reply:
fixed.

Reply:
fixed.